pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/relocation/SegmentRelocator.java

```diff
@@ -91,10 +97,66 @@ public SegmentRelocator(PinotHelixResourceManager pinotHelixResourceManager,
         Math.min(taskIntervalInMs, config.getSegmentRelocatorExternalViewCheckIntervalInMs());
     _externalViewStabilizationTimeoutInMs =
         Math.min(taskIntervalInMs, config.getSegmentRelocatorExternalViewStabilizationTimeoutInMs());
+    if (config.isSegmentRelocatorRebalanceTablesSequentially()) {
+      _waitingTables = ConcurrentHashMap.newKeySet();
+      _waitingQueue = new LinkedBlockingQueue<>();
+      _executorService.submit(() -> {
+        LOGGER.info("Rebalance tables sequentially");
+        try {
+          // Keep checking any table waiting to rebalance.
+          while (true) {
+            rebalanceWaitingTable(this::rebalanceTable);
+          }
+        } catch (InterruptedException e) {
+          LOGGER.warn("Got interrupted while rebalancing tables sequentially", e);
+          Thread.currentThread().interrupt();
```

Comment:
No need to interrupt this thread again. Remove this line and the thread will exit here
```suggestion
```